Homework3/Task1PianoKeys/Key.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task1PianoKeys
+{
+    struct Key : IComparable<Key>
+    {
+        public Notes Note;
+        public Octaves Octave;
+        public Accidentals Accidental;
+
+        public Key (Notes note, Octaves octave, Accidentals accidental = 0)
+        {
+            Note = note;
+            Octave = octave;
+            Accidental = accidental;
+
+            if(Octave == Octaves.Zero)
+            {
+                if(Note != Notes.A && Note != Notes.B)
+                {
+                    throw new ArgumentException("Octave 0 has only notes A, B");
+                }
+            }
+            else if(Octave == Octaves.Eighth)
+            {
+                if(Note != Notes.C)
+                {
+                    throw new ArgumentException("Octave 8 has only note C");
+                }
+            }
+        }
+
+        public override string ToString()
+        {
+            StringBuilder temp = new();
```

Comment:
do not use string builder here, just return interpolated string. 

StringBuilder makes sense when a lot of strings should be combined, but here single instance added.